neutrino.go

```diff
@@ -1268,6 +1279,31 @@ func (s *ChainService) handleAddPeerMsg(state *peerState, sp *ServerPeer) bool {
 		s.addrManager.Connected(sp.NA())
 	}
 
+	// We'll go through each peer subscriber and notify it about the added
+	// peer.
+	i := 0
+	for _, sub := range s.peerSubscribers {
+		select {
+		case sub.peers <- sp:
+
+		// Ignore subscription if it has been cancelled.
+		case <-sub.cancel:
+			// Avoid GC leak.
+			s.peerSubscribers[i] = nil
```

Comment:
In this case `i` is never incremented, so it'll set the same slot to nil over and over again.

Reply:
Uh, you're right. Fixed: https://github.com/lightninglabs/neutrino/pull/179/commits/c7174a06f5a2cacf6d0ea8e356fd87e3ed60ba08